api/src/main/java/org/apache/iceberg/Schema.java

```diff
@@ -586,16 +587,37 @@ private List<NestedField> reassignIds(List<NestedField> columns, TypeUtil.GetID
    * @param formatVersion table format version
    */
   public static void checkCompatibility(Schema schema, int formatVersion) {
-    // check the type in each field
+    // accumulate errors as a treemap to keep them in a reasonable order
+    Map<Integer, String> problems = Maps.newTreeMap();
```

Comment:
[optional] if we are going for another revision as fieldId() is `int` : 
```suggestion
    Map<int, String> problems = Maps.newTreeMap();
```

Reply:
Or any type parameter I think ... I feel like this is a 1337code question